pkg/k8s/factory_functions.go

```diff
@@ -126,19 +124,8 @@ func EqualV1Services(svc1, svc2 *types.Service) bool {
 		return false
 	}
 
-	clusterIP := net.ParseIP(svc1.Spec.ClusterIP)
-	headless := false
-	if strings.ToLower(svc1.Spec.ClusterIP) == "none" {
-		headless = true
-	}
-	si1 := NewService(clusterIP, svc1.Spec.ExternalIPs, headless, svc1.Labels, svc1.Spec.Selector)
-
-	clusterIP = net.ParseIP(svc2.Spec.ClusterIP)
-	headless = false
-	if strings.ToLower(svc2.Spec.ClusterIP) == "none" {
-		headless = true
-	}
-	si2 := NewService(clusterIP, svc2.Spec.ExternalIPs, headless, svc2.Labels, svc2.Spec.Selector)
+	_, si1 := ParseService(svc1)
+	_, si2 := ParseService(svc2)
 
 	// Please write all the equalness logic inside the K8sServiceInfo.Equals()
```

Comment:
Please add `if sid1 != sid2 { return false }`

Reply:
Previously, it didn't check the IDs, but OK, I will add.